crates/nu-parser/src/parser.rs

```diff
@@ -2323,6 +2323,9 @@ pub fn parse_unit_value<'res>(
         let lhs = strip_underscores(value[..lhs_len].as_bytes());
         let lhs_span = Span::new(span.start, span.start + lhs_len);
         let unit_span = Span::new(span.start + lhs_len, span.end);
+        if lhs.ends_with('$') {
+            return None;
```

Comment:
Maybe worth commenting on the reasoning on this otherwise obscure special case.

```suggestion
            // If `parse_unit_value` has higher precedence over `parse_range`,
            // a variable with the name of a unit could otherwise not be used as the end of a range.
            return None;
```